sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/serialization/CoderBasedKafkaSerializer.java

```diff
@@ -20,29 +20,79 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.common.collect.ImmutableMap;
+
+import java.io.InputStream;
 import java.util.Map;
 import javax.annotation.Nullable;
+
+import org.apache.beam.sdk.coders.AvroCoder;
 import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.io.kafka.KafkaIO;
 import org.apache.beam.sdk.util.CoderUtils;
+import org.apache.kafka.common.serialization.Deserializer;
 import org.apache.kafka.common.serialization.Serializer;
 
 /**
- * Implements Kafka's {@link Serializer} with a {@link Coder}.
+ * Implements a Kafka {@link Deserializer} and {@link Serializer} with a {@link Coder}.
```

Comment:
I removed all the references to it in KafkaIO. Even for Avro, I think Kafka has Avro deserializers. 
The primary purpose is to ease transition for users that are using non trivial coders. 

I think it is ok for now. We can certainly remove it in next update along with recipe for using Coders.  I can leave a comment in KafkaIOTest that it is just for a demo, and it is simpler to use IntegerDeserializer from Kafka directly. 

Users can always define wrapper `Deserializer` class trivially. This class is a bit involved since it is designed to work with any coder.